cpp/src/arrow/util/bit_util.h

```diff
@@ -43,13 +43,18 @@
 
 #if defined(_MSC_VER)
 #include <intrin.h>
+#include <nmmintrin.h>
 #pragma intrinsic(_BitScanReverse)
 #pragma intrinsic(_BitScanForward)
 #define ARROW_BYTE_SWAP64 _byteswap_uint64
 #define ARROW_BYTE_SWAP32 _byteswap_ulong
+#define ARROW_POPCOUNT64 __popcnt64
+#define ARROW_POPCOUNT32 __popcnt
 #else
 #define ARROW_BYTE_SWAP64 __builtin_bswap64
 #define ARROW_BYTE_SWAP32 __builtin_bswap32
+#define ARROW_POPCOUNT64 __builtin_popcountll
+#define ARROW_POPCOUNT32 __builtin_popcount
```

Comment:
TIL, C++20 has std::popcount.

Reply:
Yep, so we'll be able to use it by 2030 hopefully? :)

Reply:
I'd say 2027.